### PR TITLE
[ODS-6350] Improve Problem Details test coverage

### PR DESCRIPTION
**Pull request by axelmarquezh**

### CODE REVIEW

File: Application/EdFi.Ods.Features/ChangeQueries/Controllers/KeyChangesController.cs [hunk 1]
```diff
@@ -96,7 +96,7 @@ public class KeyChangesController : ControllerBase
             if (parameterMessages.Any())
             {
                 return BadRequest(
-                    new BadRequestException("The parameters provided to the request were invalid.", parameterMessages)
+                    new BadRequestParameterException("Parameters supplied to the request were invalid.", parameterMessages)
```
Comment (axelmarquezh) [inline Application/EdFi.Ods.Features/ChangeQueries/Controllers/KeyChangesController.cs:99]: Modified this exception to match the one in the `DeletesController`.

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Infrastructure/Configuration/NHibernateOdsConnectionProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Ods.Common.Database;
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Infrastructure/Configuration/NHibernateOdsConnectionProviderTests.cs:6]: Sort usings, with System. namespaces at the top

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Models/Domain/PropertyTypeTests.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Ods.Common.Exceptions;
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Models/Domain/PropertyTypeTests.cs:6]: Sort usings, with System. namespaces at the top

File: Application/EdFi.Ods.Tests/EdFi.Ods.Features/ExternalCache/ExternalCacheProviderTests.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Ods.Common.Exceptions;
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Features/ExternalCache/ExternalCacheProviderTests.cs:6]: Sort usings, with System. namespaces at the top

File: Postman Test Suite/Ed-Fi ODS-API Composite Base Test Suite.postman_collection.json [hunk 5]
```diff
@@ -729,6 +729,74 @@
 										"description": "Scenario: Composites support GetByKey pattern requests\r\n    Given the subject of the request is \"Staff\" where\r\n    | ParameterName | Value  | Type |\r\n    | StaffUniqueId | 207288 | Key  |    \r\n    When a GET (all) request is submitted to the \"StaffFilter\" composite with the query string filter\r\n    Then the response model should only contain the records with exact matches by id"
 									},
 									"response": []
+								},
+								{
+									"name": "GetByKey pattern using a nonexistent key",
```
Comment (simpat-jesus) [inline Postman Test Suite/Ed-Fi ODS-API Composite Base Test Suite.postman_collection.json:734]: This test covers a nonexisting key scenario but is hitting an endpoint that does not exist. Either rename the test to match what is validating or use an existing endpoint

**Review by simpat-jesus — CHANGES_REQUESTED** [review]

File: Postman Test Suite/Ed-Fi ODS-API Composite Base Test Suite.postman_collection.json (hunk 5 above)
Comment (axelmarquezh) [reply Postman Test Suite/Ed-Fi ODS-API Composite Base Test Suite.postman_collection.json:734]: The endpoint does exist, it's hitting the [Get endpoint](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/main/Application/EdFi.Ods.Features/Controllers/CompositeResourceController.cs#L83) for Composites.
It covers [this line of code](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/d697debfa7ce4cb0428c868fb7c8795fdf80b13b/Application/EdFi.Ods.Features/Composites/Infrastructure/CompositeResourceResponseProvider.cs#L125).

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Infrastructure/Configuration/NHibernateOdsConnectionProviderTests.cs (hunk 2 above)
Comment (axelmarquezh) [reply Application/EdFi.Ods.Tests/EdFi.Ods.Common/Infrastructure/Configuration/NHibernateOdsConnectionProviderTests.cs:8]: Fixed, thanks.

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Models/Domain/PropertyTypeTests.cs (hunk 3 above)
Comment (axelmarquezh) [reply Application/EdFi.Ods.Tests/EdFi.Ods.Common/Models/Domain/PropertyTypeTests.cs:7]: Fixed, thanks.

File: Application/EdFi.Ods.Tests/EdFi.Ods.Features/ExternalCache/ExternalCacheProviderTests.cs (hunk 4 above)
Comment (axelmarquezh) [reply Application/EdFi.Ods.Tests/EdFi.Ods.Features/ExternalCache/ExternalCacheProviderTests.cs:7]: Fixed, thanks.

File: Postman Test Suite/Ed-Fi ODS-API Composite Base Test Suite.postman_collection.json (hunk 5 above)
Comment (simpat-jesus) [reply Postman Test Suite/Ed-Fi ODS-API Composite Base Test Suite.postman_collection.json:734]: Oh, you are right; this is a composite; it is the same 404 exception when the endpoint does not exist.

**Review by simpat-jesus — APPROVED** [review]

**Pull request merged**